### ISSUE TITLE
Overhaul format of output

### ISSUE THREAD

**Issue by kpalmqui** [Burke-Lauenroth-Lab/SOILWAT2#21]
Overhauling the format of SOILWAT2 output files. Rather than files for each individual variable for each time period, there will now be 4 csv files (corresponding to daily, weekly, monthly, yearly time periods) that will contain all SOILWAT2 output variables.

**Comment by Zachary-Kramer** [comment]
Note for anyone who works on this: there is an old branch [soilwat_csv_output](https://github.com/Burke-Lauenroth-Lab/SOILWAT2/commits/soilwat_csv_output) that started this and may provide some guidance/ideas.

**Comment by dschlaep** [comment]
This should consider both SOILWAT2 and STEPWAT2 needs.

**Comment by BrendenBe1** [comment]
In SOILWAT there are keys for a few values (Allwthr, AllH2O, ET, and ALLVeg) that are not defined in the outsetup file and that seem to serve no purpose. There is an if statement in SW_Output.c  (below) that just checks for these and says not implemented. Is there a reason these are kept or can we remove them?

`else if ((k == eSW_AllVeg || k == eSW_ET || k == eSW_AllWthr|| k == eSW_AllH2O))
{
    SW_Output[k].use = FALSE;
    LogError(logfp, LOGNOTE, "%s : Output key %s is currently unimplemented.", MyFileName, key2str[k]);
    continue;
		}`

**Comment by Zachary-Kramer** [comment]
See issue #12

**Comment by kpalmqui** [comment]
@BrendenBe1 Should I pull down and test this standalone version of SOILWAT2 to confirm the output overhaul is working properly? Then we might be ready to close this issue?

**Comment by BrendenBe1** [comment]
@kpalmqui Ya, you should check it. I believe it is working correctly so you should test it and see if I need to fix any problems with the output.

**Comment by Zachary-Kramer** [comment]
When you do believe it is finished, please use a pull request for the merge so that others can review the code design

**Comment by BrendenBe1** [comment]
Alright, I will do that when it is complete.

**Comment by kpalmqui** [comment]
A few comments:
1) No need to output SWABULK - this has been replaced with swaGrass, swaTree, swaForb etc. 
2) Do we want to think about other names for the output files? For example, currently we have daily.csv for the non-soil layer associated daily time period variables and daily_soil.csv for the soil layer associated daily time period variables.  @Zachary-Kramer @dschlaep 
3) In the monthly output files, month 11 is repeated twice instead of month 12. In the weekly output files, week 52 is repeated twice instead of week 53.
4) For the Precip variables - in the daily.csv, they are all 0 between 1980-2010 (when historical weather is being used). Only when the markov weather is generated (year 2011) do these values get populated (and the values that are getting populated are yearly values). Also, in the monthly.csv and weekly.csv file, values are only provided for the last month/week, which are the yearly precipitation values (see yearly.csv to compare).
5) I think the daily, weekly, and monthly values are off. For example AET in 1980 in the yearly.csv file is 33.9 cm. However, when you sum monthly AET in the monthly.csv file it totals 7.96 cm. Same is true for the weekly and daily output. Summed weekly AET is 2.15 cm and summed daily AET is 0.40 cm. Something is going wrong here (unless I'm doing something wrong)?
6) Not sure that we need the soilwat_csv folder - perhaps just output these files directly into the output folder?

So a few more changes to be made. I did confirm that the current code is outputting the correct TIMESTEPs and variables upon request.

**Comment by dschlaep** [comment]
Re:

2) I agree that `daily.csv` and `daily_soil.csv` are not very helpful names -- and could easily be confused with output from STEPWAT2. What about `sw2_daily.csv` and `sw2_daily_slyrs.csv` (to indicate that these are SOILWAT2 output files and that the latter contains output for each soil layer, while still keeping the names short)?
6) I agree that we don't want an extra folder.

**Comment by BrendenBe1** [comment]
I removed the SWABULK column and fixed the week and month repetition. Should the precip values not be zero then for years 1980-2010? I am looking into the issues you found in #4 and #5 currently.

**Comment by BrendenBe1** [comment]
As for issue #5 is just added up the values that you did and they add up correctly. the sum of tall months from 1980 monthly AET is equal to the 1980 AET year. same for day and week.

**Comment by kpalmqui** [comment]
For 4. above - The precip values should not be zero. For 5. above - did you use the default outsetupv31.in? They did not add up when I tested. I will run it again and send along the output files. Thanks!

**Comment by kpalmqui** [comment]
@BrendenBe1 You are right - the daily, monthly, and weekly values are adding up correctly. Thus, only 4. remains.

I agree with @dschlaep that sw2_daily.csv and sw2_daily_slyrs.csv are good file names for the non-soil layer associated variables and soil layer associated variables, respectively.

**Comment by BrendenBe1** [comment]
@kpalmqui I was doing some more testing on the precip values and I realized that there are no ppt values being calculated for any time period other than year. I ran the master branch and its the same thing, nothing calculated but yearly. If you add the timestep year to the STEPPE outsetup.in file then the resulting sw2_yearly.csv contains the ppt values for every year.

**Comment by kpalmqui** [comment]
@BrendenBe1 Even if you request daily values in the master branch of SOILWAT2 they are not getting written? Any idea what is going wrong here @dschlaep @CaitlinA 

@CaitlinA or @dschlaep - would you be able to test the soilwat_save_output branch and confirm it is working on your end?

**Comment by dschlaep** [comment]
See also issue #60 which may overlap with some of the underlying problems described here.

I compiled and tested with commit 00877092b5a086ebb13a7e21036ca7855cc91dca:
Some problems which I noted:
- file 'sw2_daily.csv': 
    * All precip output starts with Jan 1, 2011 (when the Markov weather generator kicks in with the current setup) and has 0s before that date
    * All precip output after Jan 1, 2011 contains the same value for each day with different values among years


- file 'sw2_weekly.csv':
    * All precip output is only provided for the last week of each year until Dec 31, 2010
    * All precip output after Jan 1, 2011 contains the same value for the last week per year with different values among years

- file 'sw2_monthly.csv':
    * All precip output is only provided for December of each year until Dec 31, 2010
    * All precip output after Jan 1, 2011 contains the same value for each month with different values among years

**Comment by dschlaep** [comment]
PS I noted that branch [soilwat_save_output](https://github.com/Burke-Lauenroth-Lab/SOILWAT2/tree/soilwat_save_output) associated with this issue, is 9 commits behind master. In my experience, work is easier (particularly tasks such as preparing for final pull-request and testing) when development branches are regularly updated from master.

**Comment by BrendenBe1** [comment]
@kpalmqui Just did this and pushed the changes.
